packages/itwin/tree-widget/src/components/trees/stateless/models-tree/internal/ModelsTreeIdsCache.ts

```diff
@@ -125,11 +133,11 @@ export class ModelsTreeIdsCache {
       subjectInfos.forEach((subjectInfo, subjectId) => {
         if (subjectInfo.childModels.size > 0) {
           parentSubjectIds.add(subjectId);
-        }
-        let currParentId = subjectInfo.parentSubject;
-        while (currParentId) {
-          parentSubjectIds.add(currParentId);
-          currParentId = subjectInfos.get(currParentId)?.parentSubject;
+          let currParentId = subjectInfo.parentSubject;
+          while (currParentId) {
+            parentSubjectIds.add(currParentId);
+            currParentId = subjectInfos.get(currParentId)?.parentSubject;
+          }
         }
```

Comment:
can you check if this fixes https://github.com/iTwin/viewer-components-react/issues/928?